Add interns blog-post (#108)
vespa-engine/blog: _posts/2021-08-20-summer-interns.md

As suspected the performance of streaming search drastically decreased as
the number of concurrent users increased. We decided to change the application
to use index prefix search, and after a comparison benchmark test it was
confirmed that this implementation scaled much better than streaming search.

Like we did with the search-as-you-type application, we incorporated a static
web page java server into the Vespa application and wrote some javascript for
querying suggestions on every input and showing these suggestions in a dropdown
under the search bar.

We also took the two sample applications we made and integrated them in to an
already existing sample application
[vespa-documentation-search](https://github.com/vespa-engine/sample-apps/tree/master/vespa-cloud/vespa-documentation-search)
which is deployed on to Vespa Cloud. As of now this deployment is also used for
search suggestion on the Vespa documentation sites in the search bar.

#### AWS Lambda

One of the goals of the search-suggestion application was to favor searches

> that where previously search for. To accomplish this we decided to create a AWS

Reviewer: ```suggestion
that where previously searched for. To accomplish this we decided to create a AWS
```

Lambda function which would read query logs and feed search term from these
back into the Vespa application. The reason for this was that the query logs
where stored in AWS S3 buckets and that this would make it possible to
continuously trigger the Lambda function and process future query logs. The
biggest problem we faced when writing this Lambda function was decompressing
the logs. Vespa stores its query logs compressed with zstd-compression and
finding a zstd-library usable in a AWS Lambda context was not straight forward.
Initially a lot of time was spent learning AWS SAM and deploying a docker image
to the Lambda function, as this would let us use native C++ libraries in the
Lambda. However, we later found a library which compiled to web assembly and
would let us decompress filed with just a Node.js Lambda function.

### Other projects

While working on the main projects, we also got to work on other smaller side

> projects. One of these side project was to implement a visualization view of

Reviewer: ```suggestion
projects. One of these side projects was to implement a visualization view of
```

> [protons](https://docs.vespa.ai/en/proton.html) memory usage. This view was

Reviewer: ```suggestion
[proton's](https://docs.vespa.ai/en/proton.html) memory usage. This view was
```

During our internship at Vespa we got to learn a lot about the vespa search
engine and information retrieval in general. We have especially learned some 
different methods of doing search with incomplete queries and query processing. 
We have also gotten a feel for how it is to work in a software company through 
daily stand-ups and presentations of our projects.

Working here have given us insight into the workflow and github etiquette of a
software company. The internship has given us experience with working in a
team of developers and how to collaborate through github effectively. We
have touched upon various technologies from writing user interfaces in React to
writing performance tests in Ruby. We have also gotten to work with and learn 
about important platforms and services like Amazon Web Service and Docker, which 
is commonly used in companies but not taught in schools. 

Even after the internship there are still many things that we have not touched 
upon or learned about, and we wish we could explore more. To go deeper in 
to the lower levels of the code and the working of Vespa, and to learn more about

> search form the expirienced people working at Vespa. We grew attactched to the 

Reviewer: ```suggestion
search from the experienced people working at Vespa. We grew attached to the 
```

projects we worked on and wished we had more time to fine tune, and improve 

> them to get an even better search in the vespa documentation.

Reviewer: ```suggestion
them to get an even better search in the Vespa documentation.
```
